sdks/java/core/src/test/java/org/apache/beam/sdk/io/CompressedSourceTest.java

```diff
@@ -583,6 +586,38 @@ public void testGzipProgress() throws IOException {
   }
 
   @Test
+  public void testUnsplittable() throws IOException {
+    String baseName = "test-input";
+    File compressedFile = tmpFolder.newFile(baseName + ".gz");
+    byte[] input = generateInput(10000);
+    writeFile(compressedFile, input, CompressionMode.GZIP);
+
+    CompressedSource<Byte> source =
+        CompressedSource.from(new ByteSource(compressedFile.getPath(), 1));
+    List<Byte> expected = Lists.newArrayList();
+    for (byte i : input) {
+      expected.add(i);
+    }
+
+    PipelineOptions options = PipelineOptionsFactory.create();
+    BoundedReader<Byte> reader = source.createReader(options);
+
+    List<Byte> actual = Lists.newArrayList();
+    for (boolean hasNext = reader.start(); hasNext; hasNext = reader.advance()) {
+      actual.add(reader.getCurrent());
+      // checkpoint every 9 elements
+      if (actual.size() % 9 == 0) {
+        Double fractionConsumed = reader.getFractionConsumed();
+        if (fractionConsumed != null) {
+          assertNull(reader.splitAtFraction(fractionConsumed));
```

Comment:
when can fractionConsumed be null? This test might fail for a variety of reasons.